### PR TITLE
Refresh in widget settings

### PR DESCRIPTION
**Pull request by Negatifff**
Please give a short description what your pull request is for

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by Negatifff** [comment]
and need db correction:
alter table users_widgets modify column refresh smallint(4) NOT NULL DEFAULT '60';

but I don't know how do this in patch

**Comment by bofh80** [comment]
in:
misc\db_schema.yaml  (edit the existing schema for that table)
and:
database\migrations\newfile (copy one existing that fits the purpose and modify ).

That's if i understand correctly. :) I just checked other PR's for the same / similar.

File: app/Http/Controllers/Form/WidgetSettingsController.php [hunk 1]
```diff
@@ -45,7 +45,10 @@ public function update(Request $request, $widget_settings)
             ]);
         }
 
+        $refresh=(!is_numeric($widget_settings["refresh"]) || (int)$widget_settings["refresh"] < 0 || (int)$widget_settings["refresh"] > 32767 ? 60 : $widget_settings["refresh"]);
```
Comment (murrant) [inline app/Http/Controllers/Form/WidgetSettingsController.php:48]: This should be done in a validation rule above.

File: app/Http/Controllers/Form/WidgetSettingsController.php [hunk 2]
```diff
@@ -45,7 +45,10 @@ public function update(Request $request, $widget_settings)
...
+        unset($widget_settings["refresh"]);
```
Comment (murrant) [inline app/Http/Controllers/Form/WidgetSettingsController.php:49]: might as well unset `_token` too

File: app/Http/Controllers/Form/WidgetSettingsController.php (hunk 1 above)
Comment (murrant) [reply app/Http/Controllers/Form/WidgetSettingsController.php:48]: 'settings.refresh' => 'int|min:1|max:86400',

**Comment by murrant** [comment]
I suggest not using the "refresh" field, just put it in the settings json.
I also noticed an error in many widgets (the reason you had to set refresh in many widgets)
Inside the getSettings() function they should call parent::getSettings()

File: app/Http/Controllers/Form/WidgetSettingsController.php (hunk 1 above)
Comment (Negatifff) [reply app/Http/Controllers/Form/WidgetSettingsController.php]: ok, thanks. but I can't understand - I set not int value and after pressing save button I see empty red alert in up right corner. what I haven't do yet?

**Comment by murrant** [comment]
You won't have an empty refresh issue if you call parent::getSettings()

**Comment by Negatifff** [comment]
sorry, but I do not quite understand where to do it

**Comment by Negatifff** [comment]
@murrant thanks

**Review by murrant — APPROVED** [review]

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-69-release-changelog-october-2020/13837/1

**Pull request merged**